Have permissions be filtered (DelegatedWork or DelegatedPersonal) based on account type

lhao03 opened the pull request:
Note: This is from an intern, but I forgot to add `interns` to my branch name. 

## Overview

I noticed when using Graph Explorer as a personal account (MSA) I still get shown DelegatedWork permissions. When I try to accept them I can't. 

![delegatedPerms](https://user-images.githubusercontent.com/55033656/124133355-9b170b80-da36-11eb-8e8b-f2aa69c6b12d.gif)

### Demo
For instance, 
![image](https://user-images.githubusercontent.com/55033656/124139984-f64bfc80-da3c-11eb-8055-8772ee333052.png)
a personal account should only see "User.Read".

What this bug fix shows:
![image](https://user-images.githubusercontent.com/55033656/124139840-d288b680-da3c-11eb-9070-4b43bd1c02d2.png)

What is currently shown (only DelegatedWork):
![image](https://user-images.githubusercontent.com/55033656/124140183-1da2c980-da3d-11eb-83bc-3d68c825331f.png)

## Notes
May want to change the "Not supported" thing that pops up.
![image](https://user-images.githubusercontent.com/55033656/124140411-4f1b9500-da3d-11eb-9b66-5bc83120b9d6.png)

Also I apologize for the PR I opened and then closed when I realized there were files I didn't want to merge. 

## Testing Instructions

No new tests were added because the function `getProfileInfo` is already being tested and it calls the new function I added. Currently all existing tests pass.

thewahome commented on src/app/views/authentication/profile/Profile.tsx:
```diff
@@ -36,6 +36,13 @@ export class Profile extends Component<IProfileProps, IProfileState> {
+    const betaJsonUserInfo = actions
+      ? await actions.getProfileInfo({
+        selectedVerb: 'GET',
+        sampleUrl: BETA_USER_INFO_URL
```
Calling the Beta Profile endpoint brings in a lot of new data. Consider manipulating the data received to fit in with the IUser type and update this to carry the extra data that is needed. It seems that the object change to the profile contents stored in state breaks the `user by email` sample query.

lhao03 replied:
https://github.com/microsoftgraph/microsoft-graph-explorer-v4/pull/1008/commits/d492b75ff8a10ba86b4dba7e86ff032a355cd2b0 should fix the problem of overwriting the profile contents in state.

lhao03 commented on src/app/services/actions/profile-action-creators.ts:
```diff
@@ -43,6 +50,12 @@ export function getProfileInfo(query: IQuery): Function {
         const contentType = respHeaders['content-type'];
         const isImageResult = isImageResponse(contentType);
 
+        try {
+          dispatch(profileTypeSuccess(json.account[0].source.type[0]))
+        } catch (error) {
+          console.log(error);
+        }
+
```
Since the regular v1.0 graph URL, `${GRAPH_URL}/v1.0/me`, doesn't include the field for account type (MSA or AAD), it crashes and freezes the Graph Explorer. Maybe it's better that I create a function to check if its the beta URL, rather than doing try catch?

lhao03 replied:
Fixed in these commits, https://github.com/microsoftgraph/microsoft-graph-explorer-v4/pull/1008/commits/d492b75ff8a10ba86b4dba7e86ff032a355cd2b0, https://github.com/microsoftgraph/microsoft-graph-explorer-v4/pull/1008/commits/f0a0105d6c277638f9d4e58da8ebffaad89aaf24

lhao03 commented:
I've removed `console.log()`, the try catch and changed the code so that the profile contents in state does not get overwritten. I added a new function `isBetaURLResponse` that returns true or false and put the `profileTypeSuccess` in the existing if statement so the profile contents is not overwritten.

thewahome reviewed (approved):
This works

lhao03 commented:
Thanks for your feedback and review @thewahome! Since I don't have permissions to merge the PR, would you be able to do that if the PR is ready? Thanks 😊!
